searvey/usgs.py

```diff
@@ -214,13 +178,9 @@ def get_usgs_stations(
         symmetric=True,
     )
 
+    usgs_stations = _get_all_usgs_stations()
     if region:
-        # NOTE: Polygon is unhashable and cannot be used for a
-        # cached function input
-        region_json = mapping(region)
-        usgs_stations = _get_usgs_stations_by_region(**region_json)
-    else:
-        usgs_stations = _get_all_usgs_stations()
+        usgs_stations = usgs_stations[usgs_stations.within(region)]
```

Comment:
Nice. Not sure about the performance impact/difference, but this does simplify the code

Reply:
For single request, it's definitely slower, but if one wants to do multiple regions in the same session, it would be much faster, since the stations are cached. So it really depends on the use-case. We can go back to what we used to have if it's took slow based on user feedback.